File: 0009-Palindrome-Number/lc0009.py
# Python solution.

class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0 or (x%10 == 0 and x != 0):
            return False
        revert = 0
        while x > revert:
            revert = revert * 10 + x % 10
            x = int(x/10);
        return x == revert or x == int(revert/10)

File: 0009-Palindrome-Number/lc0009.rs
// Rust solution.

impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        if (x < 0 || (x%10 == 0 && x != 0)) { return false; }
        let mut revert: i32 = 0;
        let mut x: i32 = x;
        while (x > revert) {
            revert = revert * 10 + x % 10;
            x /= 10;
        }
        x == revert || x == revert/10
    }
}
